api/client/webclient/webclient_test.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package webclient
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestTunnelAddr(t *testing.T) {
+	testTunnelAddr := func(settings SSHProxySettings, expectedTunnelAddr string) func(*testing.T) {
```

Comment:
This looks backwards to a "normal" table-driven test to me, in that I'd normally expect to see a collection of the test case data that gets iterated over rather than multiple explicit calls to `t.Run()` as per RFD-0001 and `TestExtract()` below. 

Is there a reason for doing it this way that I'm missing?

Reply:
I started with the table version and got a bit experimentative. This version takes fewer lines and I found it to be a bit more readable, but probably better to stick to the more standard table format.

Reply:
FWIW I personally find the original tests more readable/ergonomic too (as well as more amenable to subtest parallelization). There is a short discussion/comparison at https://medium.com/@cep21/closure-driven-tests-an-alternative-style-to-table-driven-tests-in-go-628a41497e5e I haven't understood the testing guidelines RFD to _mandate_ table-driven tests in the list-of-structs form exclusively.